src/test/java/core/basesyntax/service/impl/FruitShopServiceImplTest.java

```diff
@@ -0,0 +1,97 @@
+package core.basesyntax.service.impl;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import core.basesyntax.handler.PurchaseHandlerImpl;
+import core.basesyntax.handler.SupplierHandlerImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.strategy.OperationStrategy;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FruitShopServiceImplTest {
+    private static FruitShopService fruitShopService;
+    private static OperationStrategy operationStrategyMock;
+    private static OperationHandler supplierHandlerMock;
+    private static OperationHandler purchaseHandlerMock;
+
+    @BeforeEach
+    void setUp() {
+        operationStrategyMock = mock(OperationStrategy.class);
```

Comment:
have no idea how this "mock" works, but your solution looks very good😂

Reply:
Mock object is simulated versions of real object that has the behavior of the real objects they represent. It used to isolate and test specific units of code without having to rely on the actual dependencies of that code. By using mock objects you can control the behavior of these dependencies and create predictable scenarios for testing.

Reply:
kinda super usefull thing